hardware/blebox/workarounds/colorswitch.h

```diff
@@ -0,0 +1,4 @@
+#pragma once
```

Comment:
Just drop all workarounds, use the normal functions in DomoticzHardaware.h when possible
This is crazy

Reply:
- missing includes added in a3772e3c9 (will be kept as separate commit)
- added proper cereal handling (allows using external cereal) in e34b0c69f (as separate commit)
- workarounds removed in 95f0236e7 (will be squashed/rebased later)

> use the normal functions in DomoticzHardaware.h 

I don't know what you mean. I tried to use as much as possible from existing codebase, unless changes were needed that could potentially interfere/break other integrations.

The `Send*` methods were especially problematic, with weird/inconsistent handling of fields/values and DeviceID/Unit values, that caused duplicate records to appear (which was a big problem even in the previous integration here) - this lead to lots of partially initialized records in the database, which was a horror to debug and fix. Of course, any necessary fixes could have led to breaking other integrations, which obviously wouldn't be acceptable. (Even the previous integration reimplemented one of the Send* methods for the same reason).

Reimplementing those methods to match the `decode_*` methods was safer, more robust and easier to track and test.

Overall, the Send_* methods are a bad idea overall, and the whole API should be based rather on structures for type safety (like the `_tGeneralSwitch`) to avoid mismatching Send_* methods upon switching widget types. (Which was a problem when migrating the old implementation away form pTypeLighting2).

There's a strong reason for everything I did - my primary approach was to reuse as much as possible (using "the Domoticz way" where possible) and change Domoticz internals as little as possible (ideally nothing, since other integrations could be relying on broken functionality).

If it's possible to reuse Domoticz functionality without breaking BleBox integration, that will be obvious by running the tests - so as maintainers, you can change anything you want with the confidence that if tests work, actual devices will work too.

Reply:
If you have problems with the DeviceID/Unit fields in the database (as maybe it does not fit in a 32bit value), you can also use a temporary workaround for this and use the table 'WOLNodes' for now.
This table will be renamed in a feature release to reflect a better name.
Some other hardware types are also using this table to convert a large ID to a Usable Domoticz DeviceID/Unit, with this you should be able to create enough devices
I do not like so much code for 1 device type, hope you can understand
And as mentioned, please drop the whole test folder